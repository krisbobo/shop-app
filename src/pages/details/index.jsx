import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { getProductsById } from '../../redux/detailSlice';
import {
  Container,
  ImageContainer,
  Image,
  DetailsContainer,
  Title,
  Description,
  Price,
  Category,
  Rate,
} from './styles/details';

const Details = () => {
  const dispatch = useDispatch();
  const { pathname } = useLocation();
  const detailsReducer = useSelector((state) => state.detailsReducer);
  const { singleProduct: product } = detailsReducer;
  const { isActive: isOpen } = useSelector((state) => state.homeReducer);
  const {
    category, description, image, price, rating, title,
  } = product;
  const productId = pathname.split('/')[3];

  useEffect(() => {
    dispatch(getProductsById(productId));
  }, [dispatch, productId]);

  return (
    <Container isActive={isOpen}>
      <ImageContainer>
        <Image src={image} alt={title} />
      </ImageContainer>
      <DetailsContainer>
        <Title>{title}</Title>
        <Description>{description}</Description>
        <Price>
          Price: $
          {price}
        </Price>
        <Rate>
          Rate:
          {rating?.rate}
        </Rate>
        <Category>
          Category:
          {category}
        </Category>
      </DetailsContainer>
    </Container>
  );
};

export default Details;
