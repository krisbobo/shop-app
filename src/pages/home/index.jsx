import { useSelector } from 'react-redux';
import { Products } from '../../components';
import { Main, Title } from './styles/home';

const Home = () => {
  const homeReducer = useSelector((state) => state.homeReducer);
  const { isActive: isOpen, categoryTitle, products } = homeReducer;

  return (
    <Main isActive={isOpen}>
      <Title isActive={isOpen}>
        {`${categoryTitle} (${products.length})`}
      </Title>
      <Products />
    </Main>
  );
};

export default Home;
