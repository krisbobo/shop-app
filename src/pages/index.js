export { default as Home } from './home/index';
export { default as Details } from './details/index';
