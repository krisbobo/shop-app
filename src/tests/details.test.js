import { render } from '@testing-library/react';
import { useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Details } from '../pages';

jest.mock('react-redux');
jest.mock('react-router-dom');

describe('Test Details Page components rendering', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('should render details page', async () => {
    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);
    useLocation.mockReturnValue({ pathname: '/details/product/1' });
    useSelector.mockReturnValue({
      singleProduct: {
        category: 'electronics',
        description: 'test',
        id: 14,
        image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
        price: 999.99,
        title: 'title',
        rating: {
          rate: 2.2,
          count: 140,
        },
      },
      isActive: false,
    });

    const tree = render(<Details />);
    expect(tree).toMatchSnapshot();
  });
});
