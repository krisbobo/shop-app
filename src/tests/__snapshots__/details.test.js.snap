// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Details Page components rendering should render details page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="details__Container-sc-lnyyf6-0 iswsjS"
      >
        <div
          class="details__ImageContainer-sc-lnyyf6-1 cKiZIi"
        >
          <img
            alt="title"
            class="details__Image-sc-lnyyf6-2 eyhuuM"
            src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
          />
        </div>
        <div
          class="details__DetailsContainer-sc-lnyyf6-3 eCSrtt"
        >
          <h2
            class="details__Title-sc-lnyyf6-4 cWnCgk"
          >
            title
          </h2>
          <p
            class="details__Description-sc-lnyyf6-5 jLrxXK"
          >
            test
          </p>
          <span
            class="details__Price-sc-lnyyf6-6 cRoyhv"
          >
            Price: $
            999.99
          </span>
          <span
            class="details__Rate-sc-lnyyf6-8 kfAvDS"
          >
            Rate:
            2.2
          </span>
          <span
            class="details__Category-sc-lnyyf6-7 cZTTKD"
          >
            Category:
            electronics
          </span>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="details__Container-sc-lnyyf6-0 iswsjS"
    >
      <div
        class="details__ImageContainer-sc-lnyyf6-1 cKiZIi"
      >
        <img
          alt="title"
          class="details__Image-sc-lnyyf6-2 eyhuuM"
          src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
        />
      </div>
      <div
        class="details__DetailsContainer-sc-lnyyf6-3 eCSrtt"
      >
        <h2
          class="details__Title-sc-lnyyf6-4 cWnCgk"
        >
          title
        </h2>
        <p
          class="details__Description-sc-lnyyf6-5 jLrxXK"
        >
          test
        </p>
        <span
          class="details__Price-sc-lnyyf6-6 cRoyhv"
        >
          Price: $
          999.99
        </span>
        <span
          class="details__Rate-sc-lnyyf6-8 kfAvDS"
        >
          Rate:
          2.2
        </span>
        <span
          class="details__Category-sc-lnyyf6-7 cZTTKD"
        >
          Category:
          electronics
        </span>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
